dvc/remote/s3.py

```diff
@@ -302,3 +304,57 @@ def walk_files(self, path_info, max_items=None):
                 continue
 
             yield path_info.replace(path=fname)
+
+    def _append_aws_grants_to_extra_args(self, config):
+        # Keys for extra_args can be one of the following list:
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/customizations/s3.html#boto3.s3.transfer.S3Transfer.ALLOWED_UPLOAD_ARGS
+        """
+          ALLOWED_UPLOAD_ARGS = [
+            'ACL', 'CacheControl', 'ContentDisposition', 'ContentEncoding',
+            'ContentLanguage', 'ContentType', 'Expires', 'GrantFullControl',
+            'GrantRead', 'GrantReadACP', 'GrantWriteACP', 'Metadata',
+            'RequestPayer', 'ServerSideEncryption', 'StorageClass',
+            'SSECustomerAlgorithm', 'SSECustomerKey', 'SSECustomerKeyMD5',
+            'SSEKMSKeyId', 'WebsiteRedirectLocation'
+          ]
+        """
+
+        grants = {
+            Config.SECTION_AWS_GRANT_FULL_CONTROL: {
+                "extra_args_key": "GrantFullControl",
+                "self_attr": "grant_full_control",
+            },
+            Config.SECTION_AWS_GRANT_READ: {
+                "extra_args_key": "GrantRead",
+                "self_attr": "grant_read",
+            },
+            Config.SECTION_AWS_GRANT_READ_ACP: {
+                "extra_args_key": "GrantReadACP",
+                "self_attr": "grant_read_acp",
+            },
+            Config.SECTION_AWS_GRANT_WRITE_ACP: {
+                "extra_args_key": "GrantWriteACP",
+                "self_attr": "grant_write_acp",
+            },
+        }
+        overwriting_acl_grantees = False
+        for grant_option, grant_keys in grants.items():
+            grant_value = config.get(grant_option, "")
+            if grant_value:
+                overwriting_acl_grantees = True
+                self.extra_args[grant_keys["extra_args_key"]] = grant_value
+                setattr(self, grant_keys["self_attr"], grant_value)
+
+        if overwriting_acl_grantees:
+            warning_message = (
+                "Using one of the [{0}] will overwrite default ACL "
+                "Grantees list. Default ACL Grantees list include Owner "
+                "with FULL_CONTROL permissions.".format(", ".join(grants))
+            )
+
+            logger.warning(warning_message)
+
```

Comment:
Do we actually need a warning for it? Seems like setting that in the config is a pretty explicit action already. Or are you concerned with someone cloning another's project and using these accidentally? In that case, the creator could use `config --local`, which is not tracked by git.

Reply:
I thought it would be useful to know that you are changing the default ACL by using those grants.

Reply:
@rbvesb Hm, maybe I don't quite understand the importance of this. To me, it feels too explicit to result in a warning. But if you feel like this is suitable, let's keep it, no problem.

Reply:
@rbvesb Though, thinking about it a bit more, I do think it is not needed here. Imagine you using these options and getting constantly bombarded by that warning. Let's remove it, it is better for UI.

Reply:
@efiop we have a deal!